src/utils.js

```diff
@@ -35,3 +37,16 @@ const getBrowserName = (): BrowserNames => parser.getResult().browser.name;
 const getMajor = (): number => parseInt(parser.getResult().browser.major, 10);
 
 export const isIE9 = () => getBrowserName() === 'IE' && getMajor() <= 9;
+
+export const burnItDown = (message?: BurnItDownResponse) => {
+  let timeToBurnItDown =
+    message && !message.force && message.before ? message.before - new Date().getTime() : 0;
+  if (timeToBurnItDown < 0) {
+    timeToBurnItDown = 0;
+  }
+
+  setTimeout(() => {
+    unsubscribe();
+    setBurned();
```

Comment:
Good call, I'll switch them.  I agree, worst case if unsubscribe throws an error, at least we will not be able to make more requests.